packages/blockchain-api/src/blockscout.ts

```diff
@@ -129,259 +176,149 @@ export class BlockscoutAPI extends RESTDataSource {
+            if (transfers[0].fromAddressHash.toLowerCase() === userAddress) {
+              inTransfer = transfers[0]
+              outTransfer = transfers[1]
             } else {
-              event = stableTokenTxs[0]
+              inTransfer = transfers[1]
+              outTransfer = transfers[0]
             }
-            break
-
-          // just a contract call
-          default:
-            return
-        }
+            // Find the transfer related to the queried token
+            const tokenTransfer = [inTransfer, outTransfer].find(
+              (event) => event.token === args.token
+            )
+            if (tokenTransfer) {
+              events.push({
+                type: EventTypes.EXCHANGE,
+                timestamp,
+                block,
+                amount: {
+                  // Signed amount relative to the account currency
+                  value: new BigNumber(tokenTransfer.value)
+                    .multipliedBy(tokenTransfer === inTransfer ? -1 : 1)
+                    .dividedBy(WEI_PER_GOLD)
+                    .toString(),
+                  currencyCode: tokenTransfer.token,
+                  timestamp,
+                },
+                makerAmount: {
+                  value: new BigNumber(inTransfer.value).dividedBy(WEI_PER_GOLD).toString(),
+                  currencyCode: inTransfer.token,
+                  timestamp,
+                },
+                takerAmount: {
+                  value: new BigNumber(outTransfer.value).dividedBy(WEI_PER_GOLD).toString(),
+                  currencyCode: outTransfer.token,
+                  timestamp,
+                },
+                hash,
+              })
+            }
+          }
+          break
 
-        const comment = event.input ? formatCommentString(event.input) : ''
-        const eventToAddress = event.to.toLowerCase()
-        const eventFromAddress = event.from.toLowerCase()
-        const [type, address] = resolveTransferEventType(
-          userAddress,
-          eventToAddress,
-          eventFromAddress,
-          this.getAttestationAddress(),
-          this.getEscrowAddress()
-        )
-        const timestamp = new BigNumber(event.timeStamp).toNumber() * 1000
-        events.push({
-          type,
-          timestamp,
-          block: event.blockNumber,
-          amount: {
-            // Signed amount relative to the account currency
-            value: new BigNumber(event.value)
-              .multipliedBy(eventFromAddress === userAddress ? -1 : 1)
-              .dividedBy(WEI_PER_GOLD)
-              .toString(),
-            currencyCode: event.tokenSymbol,
-            timestamp,
-          },
-          address,
-          comment,
-          hash: txhash,
-        })
+        default:
+          // Warn if anything else happens
+          console.warn(`Unhandled transfers for tx ${transferTx.transactionHash}`)
```

Comment:
Nit: maybe error here? We should probably set up alerts for this. Having an error code enum defined somewhere would help with that

Reply:
Yes!